### ISSUE TITLE
Unable to attach alb to nodes autoscaling group

### ISSUE THREAD

**Issue by olibenni** [aws-ia/terraform-aws-eks-blueprints#854]
## Description
Using aws_autoscaling_attachment to attach alb target group arn with out autoscaling group for the nodes in the eks
only works on every other apply. The apply is successful but the attachment is removed on every other apply.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions
- Module version [Required]: 4.6.2

## Reproduction Code [Required]
``` hcl
module "eks_blueprints" {
  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.6.2"

  cluster_version    = "1.21"
  vpc_id             = var.vpc_id
  private_subnet_ids = var.subnet_ids
  cluster_name       = "somecluster-eks"

  self_managed_node_groups = {
    self_mg5 = {
      node_group_name    = "self_mg5"
      instance_type      = "m5.large"
      launch_template_os = "amazonlinux2eks"
      subnet_ids         = var.worker_subnet_ids
      public_ip          = true
      # This is an invalid argument at the moment
      # target_group_arns  = [var.aws_lb_target_group_arn]
  }
}

resource "aws_autoscaling_attachment" "asg_attachment_bar" {
  autoscaling_group_name = module.eks_blueprints.self_managed_node_group_autoscaling_groups[0]
  alb_target_group_arn   = var.aws_lb_target_group_arn
}
```

## Expected behavior
Expect new nodes need be be part of our target group
The autoscaling group in the eks cluster needs to be aware of the target group so that new nodes are part of it

## Actual behavior
When we first apply, the attachment is made and things work fine.
**But in our subsequent apply, the eks_blueprint notices an attachment that isn't defined in the blueprint so it removes it.**

**Comment by askulkarni2** [comment]
@olibenni out of curiosity, what is your use-case to attach an ALB directly to the ASG this way?

**Comment by olibenni** [comment]
Good question, I wanted to add a loadbalancer for the ArgoCD server, exposing the ArgoCD web ui.
Open to alternatives.

**Comment by askulkarni2** [comment]
Gotcha. What you are trying to do will certainly work but its more common to use an [Ingress configuration](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress) for such purposes. An Ingress controller such as [AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/) makes it easier to manage Elastic Load Balancers for a Kubernetes cluster from within the cluster. Checkout our [example](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples/eks-cluster-with-external-dns) which shows you can configure a complete Ingress path with a combination of ingress-nginx, LBC, external-dns. You can repurpose this example for serving of the ArgoCD UI.

**Comment by olibenni** [comment]
Thanks for the tip! Will check this out better tomorrow (working on UTC time).

**Comment by bryantbiggs** [comment]
closed in #855
